fe/fe-core/src/main/java/com/starrocks/load/pipe/Pipe.java

```diff
@@ -489,31 +468,48 @@ private void executeTask(PipeTaskDesc taskDesc) {
         }
     }
 
+    private void checkTaskExecutionResult(PipeTaskDesc taskDesc) {
+        if (taskDesc.getFuture().isCancelled()) {
+            recordTaskError(taskDesc, "task got cancelled");
+            return;
+        } else if (!taskDesc.getFuture().isDone()) {
+            return;
+        }
+        try {
+            Constants.TaskRunState taskRunState = taskDesc.getFuture().get();
+            if (taskRunState == Constants.TaskRunState.FAILED) {
+                TaskManager tm = GlobalStateMgr.getCurrentState().getTaskManager();
+                TaskRunStatus status = tm.getTaskRunHistory().getTaskByName(taskDesc.getUniqueTaskName());
+                if (status != null) {
+                    throw new DmlException("execution failed: " + status.getErrorMessage());
+                } else {
+                    throw new DmlException("task failed with unknown status");
+                }
+            }
+            taskDesc.onFinished();
+        } catch (Throwable e) {
+            String message = e.getMessage();
+            if (LabelAlreadyUsedException.isLabelAlreadyUsed(message)) {
+                taskDesc.onFinished();
+                return;
+            }
+            recordTaskError(taskDesc, e.getMessage());
+        }
+    }
+
     public void suspend() {
         try (CloseableLock l = takeWriteLock()) {
             if (this.state == State.RUNNING) {
                 this.state = State.SUSPEND;
 
-                List<PipeFileRecord> loadingFiles = Lists.newArrayList();
                 for (PipeTaskDesc task : runningTasks.values()) {
                     if (task.isTaskRunning()) {
                         task.interrupt();
-                        loadingFiles.addAll(task.getPiece().getFiles());
                     }
                 }
-                LOG.info("suspend pipe " + this);
+                LOG.info("suspend pipe {}", this);
```

Comment:
1. So after you suspend the pipe, the state is still `LOADING`?
2. Previously, the suspend will trigger a new ingestion task? Why?